### DISCUSSION TITLE
Improve options for loadSDF

### DISCUSSION THREAD

**Discussion by ShravanTata** [bulletphysics/bullet3#3733] [General]
Currrently load sdf supports only minimal options when compared to loadURDF or createMultiBody. It would be good to expose all the options to loadSDF since it is a "more widely" used description format.

**Comment by zapplelove** [comment]
I have a urdf robot description file, but there are loops in the robot structure, so I convert it to a sdf file using the command `gz sdf -p <my_model>.urdf > <my_model>.sdf`, and I can use gazebo to load it. But when I use the pybullet, I can load it use the urdf file format instead of sdf file format.
Here is the code:
```python
import pybullet as p
import pybullet_data
import os

cheetahDIR = os.path.dirname(__file__)

physicsClient = p.connect(p.GUI)  # or p.DIRECT for non-graphical version
p.setGravity(0, 0, -9.81)

p.setAdditionalSearchPath(pybullet_data.getDataPath())
planeId = p.loadURDF("plane.urdf")

cheetahStartPos = [0, 0, 0.7]
p.setAdditionalSearchPath(cheetahDIR)
# cheetahId = p.loadURDF('cheetah/cheetah.urdf', cheetahStartPos) # This line can load the robot model without loop constraint.
cheetahId = p.loadSDF('cheetah/cheetah.sdf', cheetahStartPos) # This line cannot run properly even if I have comment the loops in the sdf file
while True:
    p.stepSimulation()
p.disconnect()
```
And I have noticed [this answer](https://github.com/bulletphysics/bullet3/issues/2124#issuecomment-466168345), does this mean that the pybullet does not support loops in a sdf file?
And if I comment loops in the sdf file, pybullet still cannot load the robot. Here are the files I used:
[cheetah.zip](https://github.com/bulletphysics/bullet3/files/5504608/cheetah.zip)
